geode-cq/src/distributedTest/java/org/apache/geode/internal/cache/tier/sockets/DurableClientTestCase.java

```diff
@@ -1446,29 +1388,31 @@ public void run2() throws CacheException {
   // Then we proceed to drain and release all locks
   // The client will then reconnect
   public class RejectClientReconnectTestHook implements CacheClientProxy.TestHook {
-    CountDownLatch reconnectLatch = new CountDownLatch(1);
-    CountDownLatch continueDrain = new CountDownLatch(1);
+    final CountDownLatch reconnectLatch = new CountDownLatch(1);
+    final CountDownLatch continueDrain = new CountDownLatch(1);
     volatile boolean clientWasRejected = false;
     CountDownLatch clientConnected = new CountDownLatch(1);
 
     public void doTestHook(String spot) {
       try {
-        if (spot.equals("CLIENT_PRE_RECONNECT")) {
-          if (!reconnectLatch.await(60, TimeUnit.SECONDS)) {
-            fail("reonnect latch was never released.");
-          }
-        } else if (spot.equals("DRAIN_IN_PROGRESS_BEFORE_DRAIN_LOCK_CHECK")) {
-          // let client try to reconnect
-          reconnectLatch.countDown();
-          // we wait until the client is rejected
-          if (!continueDrain.await(120, TimeUnit.SECONDS)) {
-            fail("Latch was never released.");
-          }
-        } else if (spot.equals("CLIENT_REJECTED_DUE_TO_CQ_BEING_DRAINED")) {
-          clientWasRejected = true;
-          continueDrain.countDown();
-        } else if (spot.equals("DRAIN_COMPLETE")) {
-
+        switch (spot) {
+          case "CLIENT_PRE_RECONNECT":
+            if (!reconnectLatch.await(60, TimeUnit.SECONDS)) {
+              fail("reonnect latch was never released.");
+            }
+            break;
+          case "DRAIN_IN_PROGRESS_BEFORE_DRAIN_LOCK_CHECK":
+            // let client try to reconnect
+            reconnectLatch.countDown();
+            // we wait until the client is rejected
+            if (!continueDrain.await(120, TimeUnit.SECONDS)) {
+              fail("Latch was never released.");
+            }
+            break;
+          case "CLIENT_REJECTED_DUE_TO_CQ_BEING_DRAINED":
+            clientWasRejected = true;
+            continueDrain.countDown();
+            break;
```

Comment:
Looks like the old code had a DRAIN_COMPLETE case that was a no-op, it doesn't look like that's being handled here.

Reply:
It is not necessary, because it will fall through and do nothing anyway.

Reply:
added default just because....